### DISCUSSION TITLE
Firebase Auth - expired API key

### DISCUSSION THREAD

**Discussion by BeeBombshell** [AntlerVC/firetable#926] [Support / Q&A] [answered]
In reference to a conversation on Discord:

Hi All.. what does one do if their api has expired (auth/api-key-expired.-please-renew-the-api-key.)

Thank you.

**Comment by BeeBombshell** [comment]
I found this resource online for the issue [here](https://stackoverflow.com/questions/51948073/firebase-authentication-expired-api-key):

1. First, go to https://console.firebase.google.com/
2. Then choose the Android project
3. Then go to SETTINGS of that project
4. First, check that: sha1 and sha256 fingerprint are there for debug apk (Necessary) and release apk (necessary if you create a signed apk)
5. There you will see a download JSON file button, click on it, and download the file.
6. Then go to Android Studio and OPEN your project.
7. Then on the left side menu, click the project subfile then find the googleservices.json file.
8. Open that file by double-clicking it
9. Now open the downloaded file, googleservices.json, and open it in notepad.
10. Now copy all the content, just select all, and copy.
11. Now go to the Android Studio, and right-click on the Google services file (in IntelliJ) and click compare with the clipboard and carefully accept all the differences into the Android Studio file.
Or you can directly remove all the content and paste the copied into the Android Studio JSON file.

DONE! NOTE: Make sure you make a copy of the file before doing this.

If this doesn't work you can try the other possible solutions listed on Stack Overflow, do let me know if you need more help and we can figure something out together.

**Reply by alikibao** [reply]
I think I should have been clearer on this.. my issue is that Rowy is not connecting to Firebase.. please see attached. Is that the same as you're describing here? 
![Screenshot 2022-11-09 195241](https://user-images.githubusercontent.com/35171600/200904494-1fc00df5-5eec-424d-b991-bcc1ed56cb3c.png)

**Reply by shamsmosowi** [reply]
Hi @alikibao 
was there any changes you made on the firebase side recently?
was it working before? or is this the first time you're using rowy on this project?

**Reply by alikibao** [reply]
Hey @shamsmosowi yes.. unfortunately, we had updated some of the API keys when we had trouble with maps.. now I'm not sure where to begin to restore Rowy.

**Reply by alta-atc** [reply]
> I found this resource online for the issue [here](https://stackoverflow.com/questions/51948073/firebase-authentication-expired-api-key):
> 
> 1. First, go to https://console.firebase.google.com/
> 2. Then choose the Android project
> 3. Then go to SETTINGS of that project
> 4. First, check that: sha1 and sha256 fingerprint are there for debug apk (Necessary) and release apk (necessary if you create a signed apk)
> 5. There you will see a download JSON file button, click on it, and download the file.
> 6. Then go to Android Studio and OPEN your project.
> 7. Then on the left side menu, click the project subfile then find the googleservices.json file.
> 8. Open that file by double-clicking it
> 9. Now open the downloaded file, googleservices.json, and open it in notepad.
> 10. Now copy all the content, just select all, and copy.
> 11. Now go to the Android Studio, and right-click on the Google services file (in IntelliJ) and click compare with the clipboard and carefully accept all the differences into the Android Studio file.
>     Or you can directly remove all the content and paste the copied into the Android Studio JSON file.
> 
> DONE! NOTE: Make sure you make a copy of the file before doing this.
> 
> If this doesn't work you can try the other possible solutions listed on Stack Overflow, do let me know if you need more help and we can figure something out together.

This answer is entirely unhelpful, it does not resolve the issue at all. Android Studio is not related to Rowy as far as I can tell, it looks like the OP just copied and pasted from stack overflow

**Comment by shamsmosowi** [comment, marked as answer]
current rowy doesnt automatically update to the latest firebase web api key when gets changed, we'll fix it in next release

**Comment by alta-atc** [comment]
Hi. I have the same issue, I have a project that I cannot access via rowy, the console is showing a similar error as above. It is unclear how I resolve this issue based on this thread.

**Comment by BeeBombshell** [comment]
Hey @alta-atc, apologies for the vague answer above. There are a few steps you'll need to perform on your end:
STEP 1: Go to this page: https://myaccount.google.com/permissions?continue=https%3A%2F%2Fmyaccount.google.com%2Fsecurity and remove Rowy access.
STEP 2: Logout in the setup section, and re-authenticate.

> NOTE: If this doesn't work, try removing the browser cookies and repeat the process. (Try using Chrome browser)

Hope this works out for you!

**Reply by alta-atc** [reply]
I followed these instructions and it did not solve the issue. I am still seeing an invalid API key error for this one rowy project (the others are fine, I just have issues with one project)

Looking at the error it seems that the API key in GCP has been changed, and this has not been reflected in wherever rowy stores the API key. 

<img width="1236" alt="Screenshot 2023-04-01 at 12 14 22 PM" src="https://user-images.githubusercontent.com/102269444/229307712-c0e775a7-7ba1-4c00-ac3b-eba9690421e5.png">
